docs/2.7.0/docs/canton/usermanual/upgrading.rst

```diff
@@ -180,6 +180,99 @@ Finally, you can ping the participant to see if the system is operational
+method `addConnection` has been renamed to `addEndpoints` to better reflect the fact that it modifies an endpoint for the sequencer.
+
+Hence, command to add a new sequencer connection to the mediator would be changed to:
+
+.. code:: bash
+
+    mediator1.sequencer_connection.modifyConnections(
+        _.addEndpoints(SequencerAlias.Default, connection)
+    )
+
+
+Unique Contract Keys Deprecation
+""""""""""""""""""""""""""""""""
+The unique-contract-keys parameters for both participant and domain nodes are now marked as deprecated.
+As of this release, the meaning and default value (true) remain unchanged.
+However, contract key uniqueness will not be available in the next major version, featuring multi-domain connectivity.
+If you are already setting this key to false explicitly (preview), this behavior will be the default one after the configuration key is removed.
+If you don't explicitly set this value to false, you are encouraged to evaluate evolving your existing applications and services to avoid relying on this feature.
+You can read more on the topic in the :ref:`documentation <canton_keys>`.
+
+Causality Tracking
+""""""""""""""""""
+An obsolete early access feature to enable causality tracking, related to preview multi-domain, was removed. If you enabled it, you need to remove the following config lines, as they will not compile anymore:
+
+.. code:: bash
+
+    participants.participant.init.parameters.unsafe-enable-causality-tracking = true
+    participants.participant.parameters.enable-causality-tracking = true
+
+Besu and Fabric drivers
+"""""""""""""""""""""""
+In order to allow for independent updates of the different components, we have  separated out the drivers into a separate jar, which needs to be loaded into a separate classpath.
+As a result, deployments that use Fabric or Besu need to additionally download the jar and place it in the appropriate directory.
+Please :ref:`consult the installation documentation <canton-enterprise-drivers>` on how to obtain this additional jar.
+
+Ledger API error codes
+""""""""""""""""""""""
+
+The error codes and metadata of GRPC errors returned as part of failed command interpretation from the Ledger API have been updated to include more information.
+Previously, most errors from the Daml engine would be given as either `GenericInterpretationError` or `InvalidArgumentInterpretationError`.
+They now all have their own codes and encode relevant information in the GRPC Status metadata.
+Specific error changes are as follows:
+* `GenericInterpretationError` (Code: `DAML_INTERPRETATION_ERROR`) with GRPC status `FAILED_PRECONDITION` is now split into:
+
+    * `DisclosedContractKeyHashingError` (Code: `DISCLOSED_CONTRACT_KEY_HASHING_ERROR`) with GRPC status `FAILED_PRECONDITION`
+    * `UnhandledException` (Code: `UNHANDLED_EXCEPTION`) with GRPC status `FAILED_PRECONDITION`
+    * `UserError` (Code: `USER_ERROR`) with GRPC status `FAILED_PRECONDITION`
```

Comment:
Unrelated: USER_ERROR and DEV_ERROR are unfortunate generic error codes for something that daml engine specific

Reply:
@samuel-williams-da is changing the codes.

Reply:
I will update them to `InterpretationUserError` and `InterpretationDevError` with their snake cases respectively.

Reply:
Yep, I reviewed his PR, will push a commit here

Reply:
The generator for these docs is on the daml repo, and has been updated in this PR
https://github.com/digital-asset/daml/pull/17225